apps/els_dap/src/els_dap_general_provider.erl

```diff
@@ -692,3 +715,27 @@ do_function_breaks(Node, Module, FBreaks, Breaks) ->
     #{Module := ModBreaks} -> Breaks#{ Module => ModBreaks#{function => FBreaks}};
     _ -> Breaks#{ Module => #{line => [], function => FBreaks}}
   end.
+
+-spec ensure_connected(node(), timeout()) -> ok.
+ensure_connected(Node, Timeout) ->
+  case is_node_connected(Node) of
+    true -> ok;
+    false ->
+      % connect and monitore project node
+      case els_distribution_server:wait_connect_and_monitor(Node, Timeout) of
+        ok -> inject_dap_agent(Node);
+        _ -> stop_debugger()
```

Comment:
This could be just `error`, right?

Reply:
In this case we could crash, and let the supervisor handle the error behavior.